Project-II-Calculator/calculator/src/components/ButtonComponents/ActionButton.js

```diff
@@ -1,2 +1,16 @@
 import React from 'react';
 import './Button.css';
+
+
+const RedButtons = props =>{
+    
+    return(
+    
+        <div className='btn func' onClick={() => props.clickHandler(props.func)}>
```

Comment:
since you are already passing the 'func' into this clickHandler function in App.js there is no need to pass it in again here inside the component. 

```
from App.js >>

<RedButtons
  func={func}
  key={func}
  clickHandler={() => this.opInput(func)}
/>
```

```
from ActionButton.js >>

<div className='btn func' onClick={() => props.clickHandler()}>
```